internal/app/server/server.go

```diff
@@ -57,5 +63,17 @@ func Run(ctx context.Context, network string, cfg *config.ServiceConfig) error {
 	pb.RegisterOpenSavesServer(s, server)
 	reflection.Register(s)
 
+	go func() {
+		select {
+		case s := <-sigs:
+			log.Infof("received signal: %v\n", s)
+		case <-ctx.Done():
```

Comment:
It's not clear when you need to cancel the context. Currently it's only in the tests, but do you have any plan to support cancellation?

Reply:
I'm supporting this here since we pass in a `ctx` to `Run`. There's no immediate plan to support this. If you prefer, I can remove it.

Reply:
sgtm